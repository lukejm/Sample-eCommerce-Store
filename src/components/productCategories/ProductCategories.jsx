import './productCategories.styles.scss';

function ProductCategories({ categories }) {


  const filterCategoryStringForKey = (category) => {
    return category.replace("'", "");
  }

  const filterCategoryStringForHTML = (category) => {
    return category.replace('-', ' ');
  }

  return (
    <div className='categories-container'>
      {categories.map(category => {
        return (
          <div className='category-container' key={filterCategoryStringForKey(category)}>
            <h2>{filterCategoryStringForHTML(category)}</h2>
            <p>Shop Now</p>
          </div>
        );
      })}
    </div>
  );
}

export default ProductCategories;