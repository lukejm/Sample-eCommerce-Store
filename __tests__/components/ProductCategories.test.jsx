import { expect, test } from 'vitest';
import renderer from 'react-test-renderer';
import ProductCategories from "../../src/components/productCategories/ProductCategories.jsx";
import * as MockData from "../../__mocks__/MockData.js";

// const keysUnfiltered = MockData.categories;
const keys = MockData.categories.map(el => el.replace("'", ""));
const wrapper = renderer.create(<ProductCategories categories={keys}/>);

test('component has 4 categories when supposed to have 4 categories', () => {
  let tree = wrapper.toJSON();
  expect(tree.children.length).toBe(4);
});

test('Negated: component has 4 categories when supposed to have 4 categories', () => {
  let tree = wrapper.toJSON();
  expect(tree.children.length).not.toBe(5);
});

test('React Component contains all given categories', () => {
  let tree = wrapper.toJSON();
  let treeArray = tree.children;
  let data = keys;
  let filtered = [];
  treeArray.forEach(treeEl => {
    if (!data.includes(treeEl.children[0].children.toString())) {
      filtered.push(treeEl.children[0].children.toString());
    }
  });
  expect(filtered.length === 0).toBe(true);
});

test('Negated: React Component contains all given categories', () => {
  let tree = wrapper.toJSON();
  let treeArray = tree.children;
  let data = keys;
  data.pop();
  let filtered = [];
  treeArray.forEach(treeEl => {
    if (!data.includes(treeEl.children[0].children.toString())) {
      filtered.push(treeEl.children[0].children.toString())
    }
  });
  expect(filtered.length === 1).toBe(true);
});

test('snapshot test', () => {
  expect(wrapper).toMatchSnapshot();
});

test('make sure testing is working', () => {
  expect(true).toBe(true);
});