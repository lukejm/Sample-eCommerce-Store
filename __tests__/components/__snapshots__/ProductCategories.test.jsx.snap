// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
<div
  className="categories-container"
>
  <div
    className="category-container"
  >
    <h2>
      electronics
    </h2>
    <p>
      Shop Now
    </p>
  </div>
  <div
    className="category-container"
  >
    <h2>
      jewelery
    </h2>
    <p>
      Shop Now
    </p>
  </div>
  <div
    className="category-container"
  >
    <h2>
      mens clothing
    </h2>
    <p>
      Shop Now
    </p>
  </div>
  <div
    className="category-container"
  >
    <h2>
      womens clothing
    </h2>
    <p>
      Shop Now
    </p>
  </div>
</div>
`;
